CabalHelper/Compile.hs

```diff
@@ -413,6 +413,9 @@ errorInstallCabal cabalVer _distdir = panic $ printf "\
  where
    sver = showVersion cabalVer
 
+helperPrepared :: Version -> IO Bool
+helperPrepared compCabalVersion = isJust <$> cachedExe compCabalVersion
```

Comment:
I don't see why h-i-e would have to do this caching, you're using ghc-mod's env. support AFAIK so that's already being taken care of.

You know frankly I think this approach causes too many problems. Without refactoring a bunch of the compilation internals and making them more well behaved this interface is just going to expose inconsistent information which helps no-one and might cause problems.

The other thing is I just _really_ don't get why you'd want to bother users about this implementation detail in the first place. All tool operations could potentially take a very long time to complete so why is cabal-helper compilation special? For example on my systems it literally takes like two seconds to compile. In the ghc-mod Emacs frontend we simply made sure that we have streaming logging output so users know what's going on and the editor doesn't just hang.